vng_api_common/authorizations/middleware.py

```diff
@@ -0,0 +1,230 @@
+from .models import Applicatie, AuthorizationsConfig, Autorisatie
+from .serializers import ApplicatieUuidSerializer
+
+
+class JWTAuth:
+    def __init__(self, encoded: str = None):
+        self.encoded = encoded
+
+    @property
+    def applicaties(self) -> Iterable[Applicatie]:
+        if self.client_id is None:
+            return []
+
+        applicaties = self._get_auth()
+
+        if not applicaties:
+            auth_data = self._request_auth()
+            applicaties = self._save_auth(auth_data)
+
+        return applicaties
+
+    @property
+    def autorisaties(self) -> models.QuerySet:
+        """
+        Retrieve all authorizations relevant to this component.
+        """
+        app_ids = [app.id for app in self.applicaties]
+        config = AuthorizationsConfig.get_solo()
+        return Autorisatie.objects.filter(
+            applicatie_id__in=app_ids, component=config.component
+        )
+
+    def _request_auth(self) -> list:
+        client = AuthorizationsConfig.get_client()
+        try:
+            response = client.list(
+                "applicatie", query_params={"clientIds": self.client_id}
+            )
+        except ClientError as exc:
+            response = exc.args[0]
+            # friendly debug - hint at where the problem is located
+            if response["status"] == 403 and response["code"] == "not_authenticated":
+                detail = _(
+                    "Component could not authenticate against the AC - "
+                    "authorizations could not be retrieved"
+                )
+                raise PermissionDenied(detail=detail, code="not_authenticated_for_ac")
+            logger.warn("Authorization component can't be accessed")
+            return []
+
+        return underscoreize(response["results"])
+
+    def _get_auth(self):
+        return Applicatie.objects.filter(client_ids__contains=[self.client_id])
+
+    @transaction.atomic
+    def _save_auth(self, auth_data):
+        applicaties = []
+
+        for applicatie_data in auth_data:
+            applicatie_serializer = ApplicatieUuidSerializer(data=applicatie_data)
+            uuid = get_uuid_from_path(applicatie_data["url"])
+            applicatie_data["uuid"] = uuid
+            applicatie_serializer.is_valid()
+            applicaties.append(applicatie_serializer.save())
+
+        return applicaties
+
+    @property
+    def payload(self) -> Optional[Dict[str, Any]]:
+        if self.encoded is None:
+            return None
+
+        if not hasattr(self, "_payload"):
+            # decode the JWT and validate it
+
+            # jwt check
+            try:
+                payload = jwt.decode(
+                    self.encoded,
+                    algorithms=["HS256"],
+                    options={"verify_signature": False},
+                    leeway=settings.JWT_LEEWAY,
+                )
+            except jwt.DecodeError:
+                logger.info("Invalid JWT encountered")
```

Comment:
I tested it with Open Zaak, the logging import and declaration of the logger are missing:

```python
import logging

logger = logging.getLogger(__name__)
```